client/components/data/store-connection/index.jsx

```diff
@@ -24,7 +24,9 @@ class StoreConnection extends React.Component {
 	}
 
 	componentWillReceiveProps( nextProps ) {
-		if ( ! isEqual( this.props, nextProps ) ) {
+		const nextState = nextProps.getStateFromStores( nextProps );
+
+		if ( ! isEqual( this.state, nextState ) ) {
 			this.removeStoreListeners( this.props.stores );
 			this.addStoreListeners( nextProps.stores );
 			this.setState( nextProps.getStateFromStores( nextProps ) );
```

Comment:
Could this be `this.setState( nextState )`, since you've already set the var `nextState` with the same value?

Reply:
Nice catch! FYI @lsl pinged me for the same defect almost at the same time lol